mwclient/client.py

```diff
@@ -676,11 +744,17 @@ def email(self, user, text, subject, cc=False):
         except errors.APIError as e:
             if e.args[0] == 'noemail':
                 raise errors.NoSpecifiedEmail(user, e.args[1])
-            raise errors.EmailError(*e)
+            raise errors.EmailError(*e)  # type: ignore[misc]
 
         return info
 
-    def login(self, username=None, password=None, cookies=None, domain=None):
+    def login(
+        self,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        cookies: Optional[Mapping[str, str]] = None,
```

Comment:
fun note: requests allows this to be an `http.cookiejar.CookieJar` as well as a dict, but we effectively don't because we do `self.connection.cookies.update(cookies)` and `CookieJar` instances do not have an `update` method.